pkg/server/sqlflowserver.go

```diff
@@ -112,3 +117,31 @@ func (s *Server) Run(req *pb.Request, stream pb.SQLFlow_RunServer) error {
 	}
 	return nil
 }
+
+// SubmitWorkflow submits an Argo workflow
+//
+// TODO(wangkuiyi): Make SubmitWorkflow return an error in addition to
+// *pipe.Reader, and remove the calls to log.Printf.
+func SubmitWorkflow(sqlProgram string, modelDir string, session *pb.Session) *pipe.Reader {
```

Comment:
Why not put this function under `pkg/workflow`? and rename to `Submit` so the caller side would use `workflow.Submit()`.

Reply:
`SubmitWorkfow ` implements `run` in the `Server` struct, and maybe `workflow.Submit` returns a workflow ID is more meaningful than `*pip.Reader`